App runs fine on dev box but fails to run at all on production box

**Issue:**
New application created, run and checked into git on server A fails to run when checked out on server B. 

On server B (development profile) the the page comes up blank and the following is in the browser console:

Uncaught Error: [$injector:modulerr] Failed to instantiate module anderschultz due to:
Error: [$injector:modulerr] Failed to instantiate module infinite-scroll due to:
Error: [$injector:nomod] Module 'infinite-scroll' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
http://errors.angularjs.org/1.5.8/$injector/nomod?p0=infinite-scroll
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:68:12
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2082:17
    at ensure (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2006:38)
    at module (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2080:14)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4617:22
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4618:40
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
http://errors.angularjs.org/1.5.8/$injector/modulerr?p0=infinite-scroll&p1=Error%3A%20%5B%24injector%3Anomod%5D%20Module%20'infinite-scroll'%20is%20not%20available!%20You%20either%20misspelled%20the%20module%20name%20or%20forgot%20to%20load%20it.%20If%20registering%20a%20module%20ensure%20that%20you%20specify%20the%20dependencies%20as%20the%20second%20argument.%0Ahttp%3A%2F%2Ferrors.angularjs.org%2F1.5.8%2F%24injector%2Fnomod%3Fp0%3Dinfinite-scroll%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A68%3A12%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2082%3A17%0A%20%20%20%20at%20ensure%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2006%3A38)%0A%20%20%20%20at%20module%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2080%3A14)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4617%3A22%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4618%3A40%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:68:12
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4640:15
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4618:40
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
    at createInjector (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4523:19)
    at doBootstrap (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:1758:20)
    at bootstrap (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:1779:12)
http://errors.angularjs.org/1.5.8/$injector/modulerr?p0=anderschultzApp&p1=Error%3A%20%5B%24injector%3Amodulerr%5D%20Failed%20to%20instantiate%20module%20infinite-scroll%20due%20to%3A%0AError%3A%20%5B%24injector%3Anomod%5D%20Module%20'infinite-scroll'%20is%20not%20available!%20You%20either%20misspelled%20the%20module%20name%20or%20forgot%20to%20load%20it.%20If%20registering%20a%20module%20ensure%20that%20you%20specify%20the%20dependencies%20as%20the%20second%20argument.%0Ahttp%3A%2F%2Ferrors.angularjs.org%2F1.5.8%2F%24injector%2Fnomod%3Fp0%3Dinfinite-scroll%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A68%3A12%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2082%3A17%0A%20%20%20%20at%20ensure%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2006%3A38)%0A%20%20%20%20at%20module%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2080%3A14)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4617%3A22%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4618%3A40%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0Ahttp%3A%2F%2Ferrors.angularjs.org%2F1.5.8%2F%24injector%2Fmodulerr%3Fp0%3Dinfinite-scroll%26p1%3DError%253A%2520%255B%2524injector%253Anomod%255D%2520Module%2520'infinite-scroll'%2520is%2520not%2520available!%2520You%2520either%2520misspelled%2520the%2520module%2520name%2520or%2520forgot%2520to%2520load%2520it.%2520If%2520registering%2520a%2520module%2520ensure%2520that%2520you%2520specify%2520the%2520dependencies%2520as%2520the%2520second%2520argument.%250Ahttp%253A%252F%252Ferrors.angularjs.org%252F1.5.8%252F%2524injector%252Fnomod%253Fp0%253Dinfinite-scroll%250A%2520%2520%2520%2520at%2520http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A68%253A12%250A%2520%2520%2520%2520at%2520http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A2082%253A17%250A%2520%2520%2520%2520at%2520ensure%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A2006%253A38)%250A%2520%2520%2520%2520at%2520module%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A2080%253A14)%250A%2520%2520%2520%2520at%2520http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A4617%253A22%250A%2520%2520%2520%2520at%2520forEach%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A321%253A20)%250A%2520%2520%2520%2520at%2520loadModules%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A4601%253A5)%250A%2520%2520%2520%2520at%2520http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A4618%253A40%250A%2520%2520%2520%2520at%2520forEach%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A321%253A20)%250A%2520%2520%2520%2520at%2520loadModules%2520(http%253A%252F%252Fec2-99-99-99-999.compute-1.amazonaws.com%253A8080%252Fbower_components%252Fangular%252Fangular.js%253A4601%253A5)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A68%3A12%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4640%3A15%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4618%3A40%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0A%20%20%20%20at%20createInjector%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4523%3A19)%0A%20%20%20%20at%20doBootstrap%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A1758%3A20)%0A%20%20%20%20at%20bootstrap%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A1779%3A12)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:68:12
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2082:17
    at ensure (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2006:38)
    at module (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:2080:14)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4617:22
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4618:40
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)
http://errors.angularjs.org/1.5.8/$injector/modulerr?p0=infinite-scroll&p1=Error%3A%20%5B%24injector%3Anomod%5D%20Module%20'infinite-scroll'%20is%20not%20available!%20You%20either%20misspelled%20the%20module%20name%20or%20forgot%20to%20load%20it.%20If%20registering%20a%20module%20ensure%20that%20you%20specify%20the%20dependencies%20as%20the%20second%20argument.%0Ahttp%3A%2F%2Ferrors.angularjs.org%2F1.5.8%2F%24injector%2Fnomod%3Fp0%3Dinfinite-scroll%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A68%3A12%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2082%3A17%0A%20%20%20%20at%20ensure%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2006%3A38)%0A%20%20%20%20at%20module%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A2080%3A14)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4617%3A22%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)%0A%20%20%20%20at%20http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4618%3A40%0A%20%20%20%20at%20forEach%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A321%3A20)%0A%20%20%20%20at%20loadModules%20(http%3A%2F%2Fec2-99-99-99-999.compute-1.amazonaws.com%3A8080%2Fbower_components%2Fangular%2Fangular.js%3A4601%3A5)
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:68:12
    at http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4640:15
    at forEach (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:321:20)
    at loadModules (http://ec2-99-99-99-999.compute-1.amazonaws.com:8080/bower_components/angular/angular.js:4601:5)

**Steps to reproduce:**

1. Create a new app.  Choose defaults for everything except use gradle build.
2. Setup MySQL with db, user name and password for prod profile.
3. Add MySQL config data to production config file.
4. Verify the app runs correctly using devl and production profile.
5. Check in app, commit & push into git repo.

Go to another box which has requisite software (node, npm, yarn, bower, etc.)

1. git clone the repo from above.
2. Setup MySQL with db, user name and password for prod profile.
3. run ./gradlew

When the app is fully started, direct your browser to the server.  The page should be blank with the error shown above in the console.

4. run ./gradlew -Pprod

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gulp_test'.
> Process 'command 'node'' finished with non-zero exit value 1

BUILD FAILED

**Related issues**

#5113 
#5114

**Work around**

As indicted in issue 5113 remove infinite-scroll and angular-loading-bar from ./src/main/webapp/app/app.module.js and application runs fine in both profiles except for the problem reported in #5114 where there is a 401 unauthorized message in the console presumably because the app is trying to serve protected content and the user is not logged in. 

**JHipster Version(s)**

4.0.1 and 4.0.2

##### **JHipster configuration**

yo jhipster:info
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
anderschultz@0.0.0 /home/jeff/anderschultz
└── generator-jhipster@4.0.2 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "4.0.2",
    "baseName": "anderschultz",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "gradle",
    "enableSocialSignIn": false,
    "rememberMeKey": "-----XXXX-----",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "clientFramework": "angular1",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

ls: no such file or directory: .jhipster/*.json

##### **Browsers and Operating System**

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 1.8.3.1
node: v6.9.5
npm: 4.1.2
bower: 1.8.0

gulp:
[20:55:13] CLI version 1.2.2
[20:55:13] Local version 3.9.1

yeoman: 1.8.5
yarn: 0.19.1

<!--
To provide all information we need, you should run `yo jhipster:info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

**Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

Client browser (any) but was using Chrome on Mac.

Devl box:

  Operating System: Fedora 25 (Workstation Edition)
  CPE OS Name: cpe:/o:fedoraproject:fedora:25
  Kernel: Linux 4.9.6-200.fc25.x86_64
  Architecture: x86-64

Prod Box:

  Operating System: CentOS Linux 7 (Core)
  CPE OS Name: cpe:/o:centos:centos:7
  Kernel: Linux 3.10.0-514.6.1.el7.x86_64
  Architecture: x86-64

- [X] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/server/templates/gitignore
+++ b/generators/server/templates/gitignore
@@ -75,7 +75,7 @@ target/
 # Gradle
 ######################
 .gradle/
-build/
+/build/
 
 ######################
 # Package Files
@@ -110,7 +110,6 @@ Desktop.ini
 ######################
 # Directories
 ######################
-/build/
 /bin/
 /deploy/
 

